### DISCUSSION TITLE
Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' 'nonce-ENb21QT9vf4dEbbGNxh+DQ=='"

### DISCUSSION THREAD

**Discussion by Sudeepkumar799** [emotion-js/emotion#2822] [Q&A] [answered]
### The Problem:

I am getting following error when I add CSP headers in my application. 

![image](https://user-images.githubusercontent.com/29960036/178702078-34846f95-7f73-4850-8fb1-224c7c00bc4b.png)
![Screenshot (101)](https://user-images.githubusercontent.com/29960036/178702497-2694237c-885c-471e-87f7-90eba1e5acf8.png)

To handle inline styles and scripts violation I am adding nonce using **csp-html-webpack-plugin**. But I am still seeing this issue.

Any help will be greatly appreciated!

**Comment by srmagura** [comment, marked as answer]
Hey @Sudeepkumar799, you should be able to fix this with the `nonce` option `createCache`. https://emotion.sh/docs/@emotion/cache#nonce

**Reply by Sudeepkumar799** [reply]
@srmagura I have implemented the `none` with `createCache` in my application. I can see nonce has been added in the application, but still it is throwing same error.

**Reply by srmagura** [reply]
OK, and you are sure the CSP's nonce matches the nonce passed to `createCache`?

If you can link me to a repository that shows the problem, I may be able to help more.

**Reply by Sudeepkumar799** [reply]
My bad @srmagura. I did not use `createCache` properly. Did some changes, now it is fixed. Thank you.

**Reply by cabaisma** [reply]
Hi @Sudeepkumar799 can share you solution in this issue? I have a similar issue. Thanks

**Reply by wishtree-dparekh** [reply]
I am facing the same issue

**Comment by howlettt** [comment]
I ran into this error when I was accidentally creating styles in the same element that was rendering `<CacheProvider>`. Moving the styles into a child component fixed the error
